plugin/pkg/auth/authenticator/token/keystone/keystone.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keystone
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"k8s.io/kubernetes/pkg/auth/user"
+	openstackcp "k8s.io/kubernetes/pkg/cloudprovider/providers/openstack"
```

Comment:
Oh.... good catch.... That will be a problem... I think we need a way to short circuit that when needed.

Really, I'm attempting to share as much of the keystone ini parsing/getting of a keystone client code as possible. If we duplicate the code in multiple places, its going to be painful to update every time we need a new authentication with keystone method (ssl certs, kerberos keytab support, etc), or when v4 ever comes out.